app/js/controllers.js

```diff
@@ -0,0 +1,139 @@
+floripaApp.controller("GMapController", ['$scope', 'uiGmapGoogleMapApi', '$location', 'searchService', function ($scope, uiGmapGoogleMapApi, $location, searchService) {
+
+    $scope.goBack = function () {
+        $location.url('/');
+    };
+
+    $scope.search = function () {
+        searchService.streetName = $scope.streetName;
+        searchService.doSearch = true;
+        $location.url('/');
+    };
+
+    uiGmapGoogleMapApi.then(function (maps) {
+
+    });
+
+    $scope.putMarker = function (mapModel, location) {
+        if (!$scope.marker) {
+            $scope.marker = new google.maps.Marker({
+                position: location,
+                map: mapModel
+            });
+        } else {
+            $scope.marker.setPosition(location);
+        }
+    }
+
+    $scope.map = {
+        // should use the one from the device, just a mock
+        center: {
+            latitude: -27.5964796,
+            longitude: -48.5205031
+        },
+        events: {
+            click: function (mapModel, eventName, originalEventArgs) {
+                var e = originalEventArgs[0];
+                $scope.geocode(e.latLng);
+                $scope.putMarker(mapModel, e.latLng);
+            },
+        },
+        zoom: 17
+    };
+
+    var geocoder = new google.maps.Geocoder();
+
+    $scope.geocode = function (latlng) {
+        geocoder.geocode({
+            'latLng': latlng
+        }, function (results, status) {
+            if (status == google.maps.GeocoderStatus.OK) {
+                if (results[1]) {
+                    $scope.streetName = results[0].address_components[1].long_name;
+                    $scope.$evalAsync();
```

Comment:
Because angular doesn't update the value of $scope.streetName everytime this callback is called since it is an event from outside angular. This call forces the evaluation of the previous assignments in the current cycle if there is still time or make sure there will be a new one.